spec/syntax.md

```diff
@@ -943,6 +943,15 @@ escaped-char = backslash ( backslash / "{" / "|" / "}" )
 backslash    = %x5C ; U+005C REVERSE SOLIDUS "\"
 ```
 
+> [!NOTE]
+> The `escaped-char` rule allows escaping some characters in places where
+> they do not need to be escaped, such as braces in a _quoted_ _literal_.
+> For example, `|foo {bar}|` and `|foo \{bar\}|` are synonymous.
+
+When programmatically emitting message syntax,
+only characters that are lexically meaningful in the current context
+SHOULD be escaped.
```

Comment:
I definitely support this, the normalized version should only have the minimum number of escapes. How about rewording where the SHOULD is though?

```suggestion
When programmatically emitting message syntax, escape sequences
SHOULD only be used for characters that are lexically meaningful in the current context.
```

The idea is to get the SHOULD to be closer to "only" rather than hanging out with "be escaped".  It just seems too easy to skim past the 'only' and end up with exactly the opposite idea.

Reply:
I'd go further and not just cover programmatic generators, but people too.

```suggestion
When writing or generating a _message_, escape sequences SHOULD NOT be used
unless required by the syntax.
That is, only escape `|` inside of _literals_ 
and `{` and `}` inside of _patterns_ or _simple messages_.
```

Reply:
Even better!

Reply:
Modified @aphillips's suggestion a bit, to ensure that we're not disallowing by omission escapes in _reserved_.

```suggestion
When writing or generating a _message_, escape sequences SHOULD NOT be used
unless required by the syntax.
That is, inside _literals_ only escape `|` 
and inside _patterns_ only escape `{` and `}`.
```